src/pages/TeamOverview.tsx

```diff
@@ -62,33 +53,56 @@ interface PageState {
 const TeamOverview = () => {
     const location = useLocation();
     const {teamId} = useParams();
-    const [pageData, setPageData] = React.useState<PageState>({});
+    const [pageData, setPageData] = React.useState<PageState>({teamLead: null, teamMembers: []});
     const [isLoading, setIsLoading] = React.useState<boolean>(true);
+    const [error, setError] = React.useState<string | null>(null);
+    const [searchTerm, setSearchTerm] = React.useState<string>('');
 
     React.useEffect(() => {
-        var getTeamUsers = async () => {
-            const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
-            const teamLead = await getUserData(teamLeadId);
+        const getTeamUsers = async () => {
+            try {
+                const {teamLeadId, teamMemberIds = []} = await getTeamOverview(teamId);
+                const teamLead = await getUserData(teamLeadId);
+                const teamMembers = await Promise.all(teamMemberIds.map(id => getUserData(id)));
```

Comment:
Using promises instead of a for loop, improving readability, and adding try-catch for error handling. If there is an error display a message so that users are aware